### PR TITLE
add update_go utility

### PR DESCRIPTION
**Pull request by nikolaishields**
Add update_go utility to update go versions in rancher image-build repositories.

### CODE REVIEW

**Review by briandowns — APPROVED** [review]

File: bin/update_go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+        -s    display the hardened-build-base version of a target repository
+        -h    show help
+
+    examples: 
+        update_go -o 1.16.3b7 -n 1.17.3b7 -r image-build-etcd"
+}
+
+get_repo() {
+    repo=$1
+    repo_path="${HOME}/go/src/github.com/rancher/${repo}"
```
Comment (galal-hussein) [inline bin/update_go:22]: can you make this path customizable, I for one have my GOPATH in different path and therefor the rke2 and k3s repos are not in that path

File: bin/update_go [hunk 2]
```diff
@@ -0,0 +1,150 @@
+#!/bin/sh
+
+set -e
+
+. ../../pkg/logger-sh/logger.sh
+. ../../pkg/stdlib-sh/stdlib.sh
+
+usage() {
+    echo "usage: update_go [onrsh]
```
Comment (briandowns) [inline bin/update_go:9]:
```suggestion
    echo "usage: $0 [onrsh]
```

File: bin/update_go [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+        -o    version     
+        -n    version
+        -r    repository (rancher/image-build-base)
+        -s    display the hardened-build-base version of a target repository
+        -h    show help
+
+    examples: 
+        update_go -o 1.16.3b7 -n 1.17.3b7 -r image-build-etcd"
```
Comment (briandowns) [inline bin/update_go:17]:
```suggestion
        $0 -o 1.16.3b7 -n 1.17.3b7 -r image-build-etcd"
```

**Comment by galal-hussein** [comment]
LGTM for me except for one nit, one question tho: are we considering creating a commit for each substitution, and also pushing a pr to each corresponding repo?

File: bin/update_go (hunk 1 above)
Comment (briandowns) [reply bin/update_go:22]: What would you suggest in this case?

Comment (nikolaishields) [reply bin/update_go:22]: Good call, will do.

**Comment by nikolaishields** [comment]
@galal-hussein @briandowns Thanks for the suggestions and taking the time to review.

**Comment by nikolaishields** [comment]
@galal-hussein As to performing vcs operations as part of this script, I was considering adding those features as part of a subsequent PR. Depending on the process we want to codify, it may even be preferred to break out the vcs operations into their own library. Do you have any thoughts/ideas on the topic?

File: bin/update_go (hunk 1 above)
Comment (nikolaishields) [reply bin/update_go:22]: We couldt look to $GOPATH env var and if it doesn't exist assume $HOME/go/src/github.com/rancher/. We could also include a flag to override the location.

Comment (briandowns) [reply bin/update_go:22]: Adding a `-p` would probably be easiest. If it's given and given with an arg, it overrides the default.

File: bin/update_go [hunk 4]
```diff
@@ -0,0 +1,150 @@
+#!/bin/sh
+
+set -e
+
+. ../../pkg/logger-sh/logger.sh
+. ../../pkg/stdlib-sh/stdlib.sh
+
+usage() {
+    echo "usage: $0 [onrsh]
+        -o    version
```
Comment (briandowns) [inline bin/update_go:10]: Bring this full block of usage all back 1 tab. Formatting in the shell is off.

**Review by galal-hussein — APPROVED** [review]

**Pull request merged**